sherlockml/config.py

```diff
@@ -0,0 +1,118 @@
+import os
+from collections import namedtuple
+
+from six.moves.configparser import ConfigParser, NoSectionError, NoOptionError
+
+
+Profile = namedtuple(
+    'Profile',
+    ['domain', 'protocol', 'client_id', 'client_secret']
+)
+
+DEFAULT_PROFILE = 'default'
+DEFAULT_DOMAIN = 'sherlockml.com'
+DEFAULT_PROTOCOL = 'https'
+
+
+def load(path):
+    """Read the SherlockML configuration from a file."""
+
+    parser = ConfigParser()
+    parser.read(str(path))
+
+    def _get(section, option):
+        try:
+            return parser.get(section, option)
+        except (NoSectionError, NoOptionError):
+            return None
+
+    profiles = {}
+
+    for section in parser.sections():
+        profiles[section] = Profile(
+            domain=_get(section, 'domain'),
+            protocol=_get(section, 'protocol'),
+            client_id=_get(section, 'client_id'),
+            client_secret=_get(section, 'client_secret')
+        )
+
+    return profiles
+
+
+def load_profile(path, profile):
+    """Read a SherlockML profile from a file."""
+    profiles = load(path)
+    try:
+        return profiles[profile]
+    except KeyError:
+        return Profile(None, None, None, None)
+
+
+def _default_configuration_path():
+    xdg_config_home = os.environ.get('XDG_CONFIG_HOME')
+    if not xdg_config_home:
+        xdg_config_home = os.path.expanduser('~/.config')
+    return os.path.join(xdg_config_home, 'sherlockml', 'configuration')
+
+
+class CredentialsError(RuntimeError):
+    pass
+
+
+def _missing_credentials(type_):
+    raise CredentialsError('No {} found'.format(type_))
+
+
```

Comment:
I'd probably drop the `_override` suffix from these arguments. It seems easier for the code that calls `resolve_profile` to use the `domain` argument when setting the domain. The fact that it overrides all other sources does not need to be made explicit in the name.

Reply:
Done.